components/script/stylesheet_loader.rs

```diff
@@ -167,18 +167,23 @@ impl FetchResponseListener for StylesheetContext {
             successful = metadata.status.map_or(false, |(code, _)| code == 200);
         }
 
+        let url = self.source.url();
         let owner = elem.upcast::<Element>().as_stylesheet_owner()
             .expect("Stylesheet not loaded by <style> or <link> element!");
+
+        document.finish_load(LoadType::Stylesheet(url));
         if owner.parser_inserted() {
             document.decrement_script_blocking_stylesheet_count();
         }
 
-        let url = self.source.url();
-        document.finish_load(LoadType::Stylesheet(url));
-
         if let Some(any_failed) = owner.load_finished(successful) {
             let event = if any_failed { atom!("error") } else { atom!("load") };
             elem.upcast::<EventTarget>().fire_event(event);
+
+            if owner.parser_inserted() {
+                document.process_pending_parsing_blocking_script_if_any();
+            }
+            document.maybe_fire_load_event();
```

Comment:
AFAICT, if the call is made before the `if let` block, the `load` event may be fired before the pending parsing blocking script is processed. If put after the block, the parser will still think the loader is blocked when resuming it with the parsing blocking script.